oauth2.go

```diff
@@ -70,12 +70,10 @@ func (z *Zoho) RefreshTokenRequest() (err error) {
 // to this function which will generate your access token and refresh tokens.
 func (z *Zoho) GenerateTokenRequest(clientID, clientSecret, code, redirectURI string) (err error) {
 	err = z.checkForSavedTokens()
-
-	z.oauth.clientID = clientID
-	z.oauth.clientSecret = clientSecret
-	z.oauth.redirectURI = redirectURI
-
 	if err == ErrTokenExpired {
+		z.oauth.clientID = clientID
```

Comment:
It feels like you just went back to the way it was, maybe I didn't explain myself clearly. All I wanted to see was the following:

```
z.oauth.clientID = clientID
z.oauth.clientSecret = clientSecret
z.oauth.redirectURI = redirectURI

err = z.checkForSavedTokens()
if err == ErrTokenExpired {
    return z.RefreshTokenRequest()
}
```

Unless there is some specific reason you went back to the original, that is what I meant.

As soon as you make the change or clarify, I will approve the merge.

Reply:
Gotcha I see. Pushed update